lib/test_prof/memory_prof/minitest.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "minitest/base_reporter"
+require "test_prof/memory_prof/printer"
+
+module Minitest
+  module TestProf
+    class MemoryProfReporter < BaseReporter # :nodoc:
+      attr_reader :tracker, :printer, :current_example
+
+      def initialize(io = $stdout, options = {})
+        super
+
+        configure_profiler(options)
+
+        @tracker = ::TestProf::MemoryProf.tracker
+        @printer = ::TestProf::MemoryProf::Printer.new(tracker)
+
+        @current_example = nil
+      end
+
+      def prerecord(group, example)
+        set_current_example(group, example)
+        tracker.example_started(current_example)
+      end
+
+      def record(example)
+        tracker.example_finished(current_example)
+      end
+
+      def start
+        tracker.start
+      end
+
+      def report
+        tracker.finish
+        printer.print
+      end
+
+      private
+
+      def set_current_example(group, example)
+        @current_example = {
+          name: example.gsub(/^test_(?:\d+_)?/, ""),
+          location: location_with_line_number(group, example)
+        }
+      end
+
+      def configure_profiler(options)
```

Comment:
In this method you rely on the symbolic keys.

That's why it should be ensured that the keys are symbolized: either by using kwargs from the beginning `**options` in line 11 (IMHO this would be preferrable because this would provide a type-like communication to the user), or by explicit symbolization right here:

```ruby
options.transform_keys!(&to_sym)
```